src/history/hs_cursor.c

```diff
@@ -322,6 +323,7 @@ __wt_hs_find_upd(WT_SESSION_IMPL *session, WT_ITEM *key, const char *value_forma
             WT_ERR_NOTFOUND_OK(__wt_hs_cursor_next(session, hs_cursor), true);
             if (ret == WT_NOTFOUND) {
                 /* Fallback to the onpage value as the base value. */
+                WT_NOT_READ(ret, 0);
```

Comment:
This seems odd - why is it necessary? We did just read the value of ret and make a decision based on it.

Reply:
Because the next code it executes after it breaks out is
```
```